### PR TITLE
v0.0.8 holiday updates

### PR DESCRIPTION
**Pull request by evancohen**
- Huge refactor (now with "plugins")
- Remote Configuration UI
- "Maker" plugin (Works with IFTTT and other home automation tech)
- Speech recognition fixes and improvements
- TV show service
- Support multiple hotwords
- Basic client side error logging
- Portuguese
- A Ton of bugfixes

**Also**
We are moving to Discord to better help serve the community. Join us:
https://discord.gg/EMb4ynW

### CODE REVIEW

File: sonus.js [hunk 1]
```diff
@@ -1,29 +1,46 @@
...
 }
 
+var keyFile = JSON.parse(fs.readFileSync(config.speech.keyFilename, "utf8"))
+var umdl = __dirname + '/node_modules/sonus/resources/snowboy.umdl'
+
 // Configure Sonus
 const Sonus = require('sonus')
 const speech = require('@google-cloud/speech')({
-  projectId: config.speech.projectId,
+  projectId: keyFile.project_id,
```
Comment (evancohen) [inline sonus.js]: I think this is stale and needs to be reverted (since we are now requiring the json file)

File: remote/config.html [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            <p id="connection-text" class="text-muted">Connecting...</p>
+        </div>
+    </footer>
+
+
+    <!-- Smart Mirror remote JavaScript
+    ================================================== -->
+    <!-- Placed at the end of the document for profit -->
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
+    <script>window.jQuery || document.write('<script src="../../assets/js/vendor/jquery.min.js"><\/script>')</script>
```
Comment (evancohen) [inline remote/config.html]: @justbill2020 are we using two versions of jquery here? What is this?

Comment (justbill2020) [inline remote/config.html]: that's also in your index.html that was a direct copy

Comment (justbill2020) [inline remote/config.html]: but yes 1.12.4 for jquery and 1.12.1 i think for jqueryui

Comment (evancohen) [inline remote/config.html]: `"../../assets/js/vendor/jquery.min.js"` doesn't exist anymore, so I'm going to get rid of it.

File: remote/index.html [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            <p id="connection-text" class="text-muted">Connecting...</p>
+        </div>
+    </footer>
+
+
+    <!-- Smart Mirror remote JavaScript
+    ================================================== -->
+    <!-- Placed at the end of the document for profit -->
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
+    <script>window.jQuery || document.write('<script src="../../assets/js/vendor/jquery.min.js"><\/script>')</script>
```
Comment (justbill2020) [inline remote/index.html]: its here too

File: remote/config.html (hunk 2 above)
Comment (justbill2020) [inline remote/config.html]: ok it should get pulled from 
/remote/index.html

**Pull request merged**